kvstore/vault.go

```diff
@@ -152,15 +157,35 @@ func valueMap(value string) map[string]interface{} {
 
 // Vault API
 
-func (c *VaultKeyValueStore) GetData(key string, version int) (map[string]interface{}, error) {
+func (c *VaultKeyValueStore) beforeRequest(ctx context.Context, path string, action string) context.Context {
+	ctx, span := c.Tracer.Start(ctx, "vault request")
+	span.SetAttributes(attribute.Key("path").String(path), attribute.Key("action").String(action))
+	return ctx
+}
+
+func (c *VaultKeyValueStore) afterRequest(ctx context.Context, err error) {
+	span := trace.SpanFromContext(ctx)
+	defer span.End()
+	if err != nil {
+		span.SetAttributes(attribute.Key("error").String(err.Error()))
```

Comment:
I think we are using `error` as a boolean, at least it seems to be the behaviour from gRPC instrumentation:
https://ui.honeycomb.io/circleci/datasets/distributed-traces/result/csFqoPCAAn